packages/SwingSet/src/controller/startXSnap.js

```diff
@@ -62,32 +88,47 @@ export function makeStartXSnap(options) {
     netstringMaxChunkSize: NETSTRING_MAX_CHUNK_SIZE,
   };
 
+  /** @param {StartXSnapLoadDetails | undefined} loadDetails */
+  function getSnapshotLoadOptions(loadDetails) {
+    if (!loadDetails) return undefined;
+
+    if (loadDetails.source === 'snapStore' && snapStore) {
+      const { vatID } = loadDetails;
+      const { hash: snapshotID, snapPos } = snapStore.getSnapshotInfo(vatID);
+      // console.log('startXSnap from', { snapshotID });
+      const snapshotStream = snapStore.loadSnapshot(vatID);
+      const snapshotDescription = `${vatID}-${snapPos}-${snapshotID}`;
+      return { snapshotStream, snapshotDescription };
+    }
+
+    if (loadDetails.source === 'snapshot') {
+      const { snapshotStream, snapshotDescription } = loadDetails;
+      return { snapshotStream, snapshotDescription };
+    }
+
+    return undefined;
```

Comment:
Is this last case an error? When would `loadDetails` be defined but not have a usable `.source`? Maybe this should throw instead.

Or.. `.source: 'bundle'` means "don't load a snapshot"? I guess that works, but I think I'd rather have the caller not provide `details.load` at all, rather than suggesting that the vat bundle (which must be processed much later, and not by `startXSnap`) is useful to appear here.

Reply:
These would be the "init" bundles, aka lockdown and supervisor, and are the ones this file explicitely deal with. I will try to make that more obvious.